entry.go

```diff
@@ -85,14 +96,46 @@ func (entry *Entry) WithFields(fields Fields) *Entry {
 	return &Entry{Logger: entry.Logger, Data: data}
 }
 
+// getCaller retrieves the name of the first non-logrus calling function
+func getCaller() (method string) {
+	// Restrict the lookback to 25 frames - if it's further than that, report UNKNOWN
+	pcs := make([]uintptr, 25)
+
+	// the first non-logrus caller is at least three frames away
```

Comment:
This will fail if the calling package is also called `logrus`, which might be common in a local project wrapper. It might be better to parameterize the number of calls from the `getCaller()` function.